org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/window/WRecordIDDialog.java

```diff
@@ -0,0 +1,220 @@
+		int tableID = tableIDValue != null ? tableIDValue.intValue() : 0;
+		recordsEditor = tableID > 0 ? new WSearchEditor("Record_ID", false, false, true, editor.getRecordsLookup(tableID)) : null;
+    	
+		setPage(page);
+		setClosable(true);
+		setTitle(Msg.getMsg(Env.getCtx(), "ChooseRelatedRecord"));
+		setPosition("center");
+		setSclass("recordid-dialog");
+		
+		contentDiv.setSclass("recordid-dialog-content");
+		confirmPanelDiv.setSclass("recordid-dialog-confirm");
+		
+		if (editor.getGridField().getGridTab() != null) {
+			parentTextBox = new Textbox();
+			parentTextBox.setReadonly(true);
+			parentTextBox.setValue(editor.getIdentifier(
+					editor.getGridField().getGridTab().getAD_Table_ID(), editor.getGridField().getGridTab().getRecord_ID()));
+		}
+		
+		tableIDEditor.getComponent().addEventListener(Events.ON_SELECT, this);
+		
+		if (recordsEditor != null)
+			recordsEditor.setValue(recordIDValue);
+		
+		okBtn.addEventListener(Events.ON_CLICK, this);
+		okBtn.setSclass("recordid-dialog-confirm");
+		cancelBtn.addEventListener(Events.ON_CLICK, this);
+		cancelBtn.setSclass("recordid-dialog-confirm");
+		
+		if (parentTextBox != null) {
+			Text text = new Text(Msg.getMsg(Env.getCtx(), "Parent"));
+			labelsDiv.appendChild(text);
+		}
+		Text text = new Text(Msg.getMsg(Env.getCtx(), "Table"));
+		labelsDiv.appendChild(text);
+		recordsEditorLabel = new Text(Msg.getMsg(Env.getCtx(), "Record"));
+		if (recordsEditor != null)
+			labelsDiv.appendChild(recordsEditorLabel);
+		labelsDiv.setSclass("recordid-dialog-labels");
+		
+		fieldsDiv.setSclass("recordid-dialog-fields");
+		
+		if (parentTextBox != null)
+			fieldsDiv.appendChild(parentTextBox);
+		fieldsDiv.appendChild(tableIDEditor.getComponent());
+		if (recordsEditor != null)
+			fieldsDiv.appendChild(recordsEditor.getComponent());
+		
+		contentDiv.appendChild(labelsDiv);
+		contentDiv.appendChild(fieldsDiv);
+		
+		confirmPanelDiv.appendChild(okBtn);
+		confirmPanelDiv.appendChild(cancelBtn);
+		
+		vLayout.appendChild(contentDiv);
+		vLayout.appendChild(confirmPanelDiv);
+		
+		appendChild(vLayout);
+		doHighlighted();
+
+		tableIDEditor.getComponent().focus();
+	}
+	
+	@Override
+	public void onEvent(Event event) throws Exception {
+		if(event.getName().equalsIgnoreCase(Events.ON_CLICK)) {
+			if(event.getTarget().equals(okBtn)) {
+				// set the selected values to the editors
+				editor.setAD_Table_ID(tableIDEditor.getValue());
+				editor.setValue(recordsEditor.getValue());
+				onClose();
+			}
+			else if(event.getTarget().equals(cancelBtn)) {
+				// dismiss the selected values
+				onClose();
+			}
+		}
+		if(event.getName().equalsIgnoreCase(Events.ON_SELECT)) {
+			if (event.getTarget().equals(tableIDEditor.getComponent())) {
+				// the Record_ID should be cleared when a different AD_Table_ID is selected
+				if (recordsEditor != null) {
+					recordsEditorLabel.detach();
+					recordsEditor.getComponent().detach();
+					editor.setValue(null);
```

Comment:
this is not right, should only happens when user press the apply button.

Reply:
ok thanks, then probably best is to clear recordsEditor here, since that value is what editor.setValue(..); uses on OK clicked